### PR TITLE
Infrastructure Container Refactor

### PR DESCRIPTION
**Pull request by poftadeh**
Did a rewrite of the Infrastructure Container on the backend. Changes:

- [x] Clean up logic and create a service file
- [x] Add ES6+ syntax and comply with standards 
- [x] Fix Promises
- [x] Create a config file for servers, workstations, dns, and sshUser
- [x] Fix frontend property names and dummy file to work with new config data
- [x] Add timeout to ssh server query
- [x] Remove `node-cmd` dependency (uses native `exec` instead)
- [ ]  Confirm that #73 is fixed. (EHL seems unreachable at the moment.)

Tldr: Fix/Progress towards #73

### CODE REVIEW

File: Dashboard/src/components/Infrastructure.js [hunk 1]
```diff
@@ -48,8 +48,8 @@ class Infrastructure extends Container {
                       key={"workstation-" + i}
                       className="infra-box"
                     >
-                      <img className="ip-icon" src={server.Status === "up" ? zap : zapOff} alt={"IP icon"} />
-                      <span> {server.Host} </span>
+                      <img className="ip-icon" src={server.status === "up" ? zap : zapOff} alt={"IP icon"} />
+                      <span> {server.host} </span>
```
Comment (belavina) [inline Dashboard/src/components/Infrastructure.js]: I see a lot of repetition, this block is repeated 3 times & looks almost identical:
```
<div className="infra-wrapper">
   <div className="infra-column">
         <div className="infra-box"><h3>{Title}</h3></div>
     ...
       </div>
 </div>
```
I think it could be moved to a function outside of render & you could pass dns, workstations & servers to it

File: Infrastructure/config.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  workstations: [
+    { host: 'bbb', ipAddress: 'bbb.cdot.systems' },
+    { host: 'vega', ipAddress: 'vega.cdot.systems' },
+    { host: 'rob', ipAddress: 'rob.cdot.systems' },
+    { host: 'ps', ipAddress: 'ps.cdot.systems' },
+    { host: 'italy', ipAddress: 'italy.cdot.systems' },
+    { host: 'canada', ipAddress: 'canada.cdot.systems' },
+    { host: 'narnia', ipAddress: 'narnia.cdot.systems' },
+    { host: 'spain', ipAddress: 'spain.cdot.systems' },
+    { host: 'rome', ipAddress: 'rome.cdot.systems' },
```
Comment (belavina) [inline Infrastructure/config.js]: these are not ipAddresses, maybe we could switch to `hosts`?

File: Dashboard/src/components/Infrastructure.js [hunk 3]
```diff
@@ -14,10 +14,10 @@ class Infrastructure extends Container {
...
 
     if (this.state.data) {
-      const upSort = (a, b) => a.Status === 'up';
-      servers = this.state.data.Servers.sort(upSort);
-      workstations = this.state.data.Workstations.sort(upSort);
-      dns = this.state.data.DNS;
+      const upSort = (a, b) => a.status === 'up';
+      servers = this.state.data.servers.sort(upSort);
+      workstations = this.state.data.workstations.sort(upSort);
+      dns = this.state.data.dns;
```
Comment (belavina) [inline Dashboard/src/components/Infrastructure.js]: Sort is an in-place algorithm, you probably need to use spread operators here first.
see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort

File: Dashboard/src/components/Infrastructure.js (hunk 1 above)
Comment (poftadeh) [reply Dashboard/src/components/Infrastructure.js]: I didn't touch the structure of the frontend since that is covered in #65. The only changes I made here were to line up with the backend changes to the data structure.

File: Dashboard/src/components/Infrastructure.js (hunk 3 above)
Comment (poftadeh) [reply Dashboard/src/components/Infrastructure.js]: See above note about #65.

File: Infrastructure/config.js (hunk 2 above)
Comment (poftadeh) [reply Infrastructure/config.js]: Changed in `dummy-data.js`, `config.js`, and `checkService.js`

File: Infrastructure/checkService.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+  dnsQueryDomain,
+  sshUser
+} = require('./config.js');
+
+const DIG_ARG = 'ns';
+const SSH_TIMEOUT = 3;
+
+const pingWorkstation = workstation => {
+  return new Promise(resolve => {
+    ping.sys.probe(workstation.address, isAlive => {
```
Comment (scrust) [inline Infrastructure/checkService.js]: ping seems to have [a promise-based option](https://www.npmjs.com/package/ping#promise-wrapper) as well, though that may not be as useful since the result needs to be mapped anyways.

File: Infrastructure/checkService.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+const exec = require('child_process').exec;
+const dig = require('node-dig-dns');
+const {
+  servers: serverList,
+  workstations: workstationList,
+  dnsQueryDomain,
+  sshUser
+} = require('./config.js');
+
+const DIG_ARG = 'ns';
```
Comment (scrust) [inline Infrastructure/checkService.js]:
```suggestion
const DIG_ARGS = ['ns'];
```

This could then be [spread](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax) (e.g. `[dnsQueryDomain, ...DIG_ARGS]`) to support multiple args in future.

Otherwise it's probably ok to inline these arguments if they're only being used once.

File: Dashboard/src/debug/dummy-data.js [hunk 6]
```diff
@@ -90,74 +90,74 @@ export default {
...
+        status: 'up'
       }
     ]
   },
   github: [
     {
       author: { name: 'Alpha Zero', date: '6/4/2019, 6:24:40 PM' },
-      repoName: 'ostep-dashboard',
-      branchName: 'voltage-fluct-57',
+      reponame: 'ostep-dashboard',
```
[the remark this answers is not in the document]
Comment (poftadeh) [reply Dashboard/src/debug/dummy-data.js]: That's right, it must have got caught in the replace-all I did for `Name`. Fixed in latest commit.

File: Infrastructure/checkService.js (hunk 4 above)
Comment (poftadeh) [reply Infrastructure/checkService.js]: I think it looks cleaner promise-based. Added the changes.

File: Infrastructure/checkService.js (hunk 5 above)
Comment (poftadeh) [reply Infrastructure/checkService.js]: That makes sense. I've included the use of spread syntax to the args.
